src/Traits/MagicPropertyTrait.php

```diff
@@ -19,6 +19,32 @@
 /**
  * Provides the magic methods that allow read access
  * to magic properties.
+ *
+ * @property-read int year
```

Comment:
Oh, I understand, and I agree with you even if I don't actually know what conventions do you refer to, but if it's about homogenize code and making it consistent, I'll alway agree. First thing I came in mind, seeing the recent changes on the framework, would be to have getters methods (may be magic ones) instead of properties, but let me know what's your idea.